libdnf/module/module_sack.cpp

```diff
@@ -478,6 +482,149 @@ std::pair<std::vector<std::vector<std::string>>, ModuleSack::ModuleErrorType> Mo
+            if (eq_position == std::string::npos) {
+                throw std::runtime_error("Invalid format (missing '=') of PLATFORM_ID in " + os_release_path);
+            }
+            auto start_position = line.find('"', eq_position + 1);
+            if (start_position == std::string::npos) {
+                throw std::runtime_error("Invalid format (missing '\"' in value) of PLATFORM_ID in " + os_release_path);
+            }
+            auto colon_position = line.find(':', eq_position + 1);
+            if (colon_position == std::string::npos) {
+                throw std::runtime_error("Invalid format (missing ':' in value) of PLATFORM_ID in " + os_release_path);
+            }
+            auto end_position = line.find('"', colon_position + 1);
+            if (end_position == std::string::npos) {
+                throw std::runtime_error("Invalid format (missing '\"' in value) of PLATFORM_ID in " + os_release_path);
+            }
+
+            return make_pair(
+                line.substr(start_position + 1, colon_position - start_position - 1),
+                line.substr(colon_position + 1, end_position - colon_position - 1));
+        }
+    }
+    return {};
+}
+
+
+static std::set<std::string> get_strings_from_provide(
+    const libdnf::rpm::PackageSet & packages, const char * provide_name) {
+    std::set<std::string> strings;
+
+    auto provide_name_length = strlen(provide_name);
+    for (auto const & package : packages) {
+        auto const & provides = package.get_provides();
+        auto found_provide = std::find_if(provides.begin(), provides.end(), [&](const libdnf::rpm::Reldep & dep) {
+            return !strncmp(dep.get_name(), provide_name, strlen(provide_name));
+        });
+        if (found_provide != provides.end()) {
+            auto found_provide_name = (*found_provide).get_name();
+            auto found_provide_name_length = strlen(found_provide_name);
+            if (found_provide_name[provide_name_length] == '(' &&
+                found_provide_name[found_provide_name_length - 1] == ')') {
+                strings.emplace(
+                    found_provide_name + provide_name_length + 1, found_provide_name_length - provide_name_length - 2);
+            }
+        }
+    }
+
+    return strings;
+}
+
+
+std::pair<std::string, std::string> ModuleSack::Impl::detect_platform_name_and_stream() const {
+    // try to detect platform id from configuration
+    auto & config_platform_option = base->get_config().module_platform_id();
+    if (!config_platform_option.empty()) {
+        auto & config_platform = config_platform_option.get_value();
+        auto parsed_platform_id = parse_platform_id_from_string(config_platform);
+        if (!parsed_platform_id.first.empty() && !parsed_platform_id.second.empty()) {
+            return parsed_platform_id;
+        } else {
+            base->get_logger()->warning("Invalid format of platform ID in configuration: {}", config_platform);
```

Comment:
Minor: The message could contain the name of configuration option.

Reply:
Definitely

Reply:
Report it as debug messages is a very good suggestion.

Reply:
Logging level was changed to debug and regex format was added for the `module_platform_id` option.